app/src/main/java/com/z1/pokedex/core/network/service/googleauth/GoogleAuthClientImpl.kt

```diff
@@ -0,0 +1,92 @@
+package com.z1.pokedex.core.network.service.googleauth
+
+import android.content.Context
+import android.content.Intent
+import android.content.IntentSender
+import com.google.android.gms.auth.api.identity.BeginSignInRequest
+import com.google.android.gms.auth.api.identity.BeginSignInRequest.GoogleIdTokenRequestOptions
+import com.google.android.gms.auth.api.identity.SignInClient
+import com.google.firebase.Firebase
+import com.google.firebase.auth.GoogleAuthProvider
+import com.google.firebase.auth.auth
+import com.z1.pokedex.BuildConfig
+import com.z1.pokedex.R
+import com.z1.pokedex.core.network.model.google.SignResult
+import com.z1.pokedex.core.network.model.google.UserData
+import kotlinx.coroutines.tasks.await
+import java.util.concurrent.CancellationException
+
+class GoogleAuthClientImpl(
+    private val context: Context,
+    private val oneTapClient: SignInClient
+): GoogleAuthClient {
+    private val auth = Firebase.auth
+
+    override suspend fun signIn(): IntentSender? {
+        val result = try {
+            oneTapClient.beginSignIn(buildSignInRequest()).await()
+        } catch (e: Exception) {
+            e.printStackTrace()
+            if (e is CancellationException) throw e
```

Comment:
pq precisa relançar a exceptions se for esse tipo em específico? Vi em vários lugares isso